runtime/jvmti/jvmtiStartup.c

```diff
@@ -1110,6 +1279,88 @@ isAgentLibraryLoaded(J9JavaVM *vm, const char *library, BOOLEAN decorate)
+		 */
+		PORT_ACCESS_FROM_JAVAVM(vm);
+		/* retrieve a list of all object files loaded for the current process */
+		struct ld_info *objectFiles = loadqueryGetObjectFiles(vm);
+		if (NULL != objectFiles) {
+			char *libNameSo = NULL;
+			char *libNameAr = NULL;
+			char *libpath = NULL;
+			UDATA libpathSize = 0;
+			char *envLibPath = NULL;
+			char *fileName = NULL;
+			BOOLEAN isLibraryPath = FALSE;
+			struct ld_info *cursor = NULL;
+
+			if (decorate) {
+				/* the library is a platform independent name */
+				UDATA libraryLen = strlen(library);
+				/* either object suffix is acceptable */
+				UDATA sizeLibNameSo = libraryLen + sizeof("lib.so");
+				UDATA sizeLibNameAr = libraryLen + sizeof("lib.a");
+				libNameSo = (char *)j9mem_allocate_memory(sizeLibNameSo, J9MEM_CATEGORY_JVMTI);
+				libNameAr = (char *)j9mem_allocate_memory(sizeLibNameAr, J9MEM_CATEGORY_JVMTI);
+				if ((NULL != libNameSo) && (NULL != libNameAr)) {
+					j9str_printf(PORTLIB, libNameSo, sizeLibNameSo, "lib%s.so", library);
+					j9str_printf(PORTLIB, libNameAr, sizeLibNameAr, "lib%s.a", library);
+					Trc_JVMTI_isAgentLibraryLoaded_libNames(libNameSo, libNameAr);
+				} else {
+					Trc_JVMTI_isAgentLibraryLoaded_OOM("libNameSo or libNameAr");
+					vm->internalVMFunctions->setNativeOutOfMemoryError(vm->mainThread, 0, 0);
+					goto failed;
+				}
+			} else {
+				/* check if library contains a path separator */
+				isLibraryPath = (NULL != strrchr(library, '/'));
+			}
+
+			/* Because redirector adds onto LIBPATH, fetching LIBPATH via getenv is more
+			 * accurate than fetching LIBPATH from loadquery using L_GETLIBATH flag.
+			 */
+			envLibPath = getenv("LIBPATH");
+			if (NULL != envLibPath) {
+				libpathSize = strlen(envLibPath) + 1;
+				libpath = j9mem_allocate_memory(libpathSize, J9MEM_CATEGORY_JVMTI);
+				if (NULL == libpath) {
+					Trc_JVMTI_isAgentLibraryLoaded_OOM("libpath");
+					vm->internalVMFunctions->setNativeOutOfMemoryError(vm->mainThread, 0, 0);
+					goto failed;
+				}
+				/* copy it to avoid the modification by putenv() */
+				memcpy(libpath, envLibPath, libpathSize);
+			} else {
+				/* assign it an empty string */
+				libpath = (char *)"";
+			}
+			Trc_JVMTI_isAgentLibraryLoaded_loadquery_getlibpath(libpath, libpathSize);
+			for (cursor = objectFiles;;) {
+				/* objectFiles is not NULL */
+				if (0 == cursor->ldinfo_next) {
+					/* last entry, exit */
+					break;
+				}
+				fileName = cursor->ldinfo_filename;
+				Trc_JVMTI_isAgentLibraryLoaded_ldinfo_filename_decorate(fileName, decorate);
+				result = searchObjectFileFromLDInfo(fileName, library, decorate, isLibraryPath, libNameSo, libNameAr, libpath);
+				if (result) {
+					break;
+				}
+				cursor = (struct ld_info *)((char *)cursor + cursor->ldinfo_next);
+			}
+failed:
+			if (libpathSize > 0) {
+				j9mem_free_memory(libpath);
+			}
+			j9mem_free_memory(libNameSo);
+			j9mem_free_memory(libNameAr);
+		}
+		j9mem_free_memory(objectFiles);
```

Comment:
This can move before line 1362 - the end of the block which is the only scope where it might be non-null.

Reply:
Moved